src/main/java/com/revature/models/TripDTO.java

```diff
@@ -3,18 +3,39 @@
 import java.time.LocalDateTime;
 import java.util.List;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
 @NoArgsConstructor @Data
 public class TripDTO {
+
 	private int tripId;
+
+	@NotNull
+	@Pattern(regexp = "[a-zA-Z0-9 ]+", message = "Trip names may only contain letters, numbers, and spaces")
 	private String name;
+
+	@NotNull
 	private User driver;
```

Comment:
This is a curious consideration, but should the nested entities also be DTOs for the DTO models? For example, TripDTO would have UserDTO and Trip would have User. That way, we can still perform JSR validation on the nested dependencies.